src/Manager.js

```diff
@@ -1,27 +1,61 @@
 import User from './User'
 
 class Manager extends User {
-  constructor(date, rooms, bookings) {
-    super(id);
-    this.date = date;
+  constructor(customers, rooms, id, date, bookings) {
+    super(id, date, bookings);
+    this.customers = customers;
     this.rooms = rooms;
+    this.currentCustomer = {};
+  }
 
+  determineOccupiedRooms(date, bookingData) {
```

Comment:
Reduce totally works. If you ever want to consider refactoring, I think a simple map would work as well. But the first priority is code that works!!